README.md

```diff
@@ -1,2 +1,81 @@
+
+
+
+## Problem
+
+
+
+For a kubernetes cluster to remain in a healthy state, all the nodes should remain in a healthy, running state.
+
+
+## Solution
+
+
+
+- Machine auto healer operator will always try to keep the nodes (machines)
+  in your cluster in a healthy, running state.
+- It will perform periodic checks on the health state of each node (machine) in your cluster.
+- If a node (machine) fails consecutive health checks over an extended time period,
+  it will initiate a repair process for that node (machine).
+
+
+
+![](./docs/images/machine_auto_healer.png)
+
+##### Node Conditions
+- The conditions field describes the status of all Running nodes.
+- By describing any node we can see the NodeCondition and its respective status
+  ```bash
+  Conditions:
+  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
+  ----             ------  -----------------                 ------------------                ------                       -------
+  MemoryPressure   False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasSufficientMemory   kubelet has sufficient memory available
+  DiskPressure     False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasNoDiskPressure     kubelet has no disk pressure
+  PIDPressure      False   Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:44:18 +0530   KubeletHasSufficientPID      kubelet has sufficient PID available
+  Ready            True    Sat, 10 Apr 2021 02:23:21 +0530   Fri, 09 Apr 2021 15:45:45 +0530   KubeletReady                 kubelet is posting ready status
+
+  ```
+- Following are the NodeConditions by default supported by K8s cluster.
+  
+  
+  | ConditionType      | Condition Status   |Effect        | Key      |
+  | ------------------ | ------------------ | ------------ | -------- |
+  |Ready               |True                | -            | |
+  |                    |False               | NoExecute    | node.kubernetes.io/not-ready           |
+  |                    |Unknown             | NoExecute    | node.kubernetes.io/unreachable         |
+  |OutOfDisk           |True                | NoSchedule   | node.kubernetes.io/out-of-disk         |
+  |                    |False               | -            | |
+  |                    |Unknown             | -            | |
+  |MemoryPressure      |True                | NoSchedule   | node.kubernetes.io/memory-pressure     |
+  |                    |False               | -            | |
+  |                    |Unknown             | -            | |
+  |DiskPressure        |True                | NoSchedule   | node.kubernetes.io/disk-pressure       |
+  |                    |False               | -            | |
+  |                    |Unknown             | -            | |
+  |NetworkUnavailable  |True                | NoSchedule   | node.kubernetes.io/network-unavailable |
```

Comment:
One node can have not good several conditions at one time.
Maybe need to have new condition types by mix of the basic condition types?

Reply:
We can have either a `ConfigMap` or new resource type as `ConditionSet`  which could be collection of one or more condtions.
I have mentioned 2 approachs [here](https://github.com/stakater/jarvis/blob/add-readme/docs/node-conditions.md#one-node-can-have-not-good-several-conditions-at-one-time-we-need-to-new-condition-types-that-could-be-combination-multiple-condition-types), please suggest the preferred one

Reply:
I suggest configmap because the conditions are not objects also.

Reply:
Yes, conditions are not objects but [ConditionSet](https://github.com/stakater/jarvis/blob/0.0.1/api/v1alpha1/conditionset_types.go#L45) can be an object with conditions, taint effect, taint key etc as peroperties. Or should we rename it to NodeConditionSet.
Please suggest, If `ConditionSet` doesn't make sence as an object, I will revert back and use it as a configmap.